src/DataTypes/Serializations/SerializationNumber.cpp

```diff
@@ -136,7 +137,29 @@ void SerializationNumber<T>::deserializeBinary(IColumn & column, ReadBuffer & is
 template <typename T>
 void SerializationNumber<T>::serializeBinaryBulk(const IColumn & column, WriteBuffer & ostr, size_t offset, size_t limit) const
 {
+    if (std::string(column.getFamilyName()) == "Function")
```

Comment:
So you want to postpone lazy expression execution until serialization, it's not a good solution. it's just a really strange trick that won't work properly.